### PR TITLE
Method Expression mutator

### LINKED ISSUE #1907: Function chain
**pipe/chain functions.**
It is popular and very widespred to use pipe functions or some chains.
 

>  mycollection.filter(e=>e.active).map(e=>e.value).sort();

And very usefull to remove one call from chain to verify tests

> mycollection.map(e=>e.value).sort();

> mycollection.filter(e=>e.active).sort();

> mycollection.filter(e=>e.active).map(e=>e.value);

May be it's not good for some projects and can generate some lifeless mutants,
 but my experience show, that this is one of usefull mutator,
 when you use array pipes or work with RxJs

I'm using this implementation
  https://github.com/wolf-off/stryker-rxjs/blob/master/src/mutator/FunctionChainMutator.ts

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
Adds a new `MethodExpression` mutator.

Fixes #1907.

Co-authored-by: @nicojs

### CODE REVIEW

**Comment by nicojs** [comment]
I would expect some e2e tests to fail. A lot of e2e tests use snapshots of their respective JSON reports. They should change for projects where we use the method calls.

You should be able to run `npm run test:update` from the `e2e` directory to update all snapshots (be sure to do that when you have 10 minutes to spare, as it takes some time).

File: packages/instrumenter/src/mutators/mutate.ts [hunk 1]
```diff
@@ -24,6 +25,7 @@ export const allMutators: NodeMutator[] = [
   conditionalExpressionMutator,
   equalityOperatorMutator,
   logicalOperatorMutator,
+  methodExpressionMutator,
```
Comment (JoshuaKGoldberg) [inline packages/instrumenter/src/mutators/mutate.ts:28]: Moving https://github.com/stryker-mutator/stryker-js/pull/3508#issuecomment-1136856947 to a thread...

Err:

```
$ npm run test:update

> e2e-tests@0.0.0 test:update
> cross-env CHAI_JEST_SNAPSHOT_UPDATE_ALL=true npm test


> e2e-tests@0.0.0 test
> node tasks/run-e2e-tests.js

node:internal/process/esm_loader:94
    internalBinding('errors').triggerUncaughtException(
                              ^

Error [ERR_UNSUPPORTED_DIR_IMPORT]: Directory import '/home/josh/repos/stryker-js/node_modules/rxjs/operators' is not supported resolving ES modules imported from /home/josh/repos/stryker-js/e2e/tasks/run-e2e-tests.js
Did you mean to import rxjs/operators/index.js?
    at new NodeError (node:internal/errors:371:5)
    at finalizeResolution (node:internal/modules/esm/resolve:414:17)
    at moduleResolve (node:internal/modules/esm/resolve:981:10)
    at defaultResolve (node:internal/modules/esm/resolve:1078:11)
    at ESMLoader.resolve (node:internal/modules/esm/loader:530:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:251:18)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:79:40)
    at link (node:internal/modules/esm/module_job:78:36) {
  code: 'ERR_UNSUPPORTED_DIR_IMPORT',
  url: 'file:///home/josh/repos/stryker-js/node_modules/rxjs/operators'
}
```

Comment (nicojs) [reply packages/instrumenter/src/mutators/mutate.ts:28]: Weird! Did you `npm install` inside `e2e` first?

Comment (JoshuaKGoldberg) [reply packages/instrumenter/src/mutators/mutate.ts:28]: Oh! 🤦 that did the trick. And now I see the `npm run e2e` note in `CONTRIBUTING.md`. Great! Running now...

**Pull request merged**